PROJECT

```diff
@@ -32,6 +32,10 @@ resources:
   kind: ByoCluster
   path: github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/apis/infrastructure/v1beta1
   version: v1beta1
+  webhooks:
+    defaulting: true
```

Comment:
Keep this for future. It would be save our energy if we want add something for Default.

Reply:
I'm also actually +1 for removing this. Let's add it only when needed. Keeping only things that we need now will reduce maintenance overhead.

Reply:
But if we want to  auto-generate the default function again, we need to delete the whole ByoCluster webhook first. Or it will report the following error:
```
2022/01/11 02:44:11 failed to create webhook: unable to inject the resource to "base.go.kubebuilder.io/v3": webhook resource already exists
```
This will cause extra burden。

Reply:
At that time, we only write this new function. It doesn't necessarily have to be generated via kubebuilder. I'm stressing on removing this because this is the code without any tests and no usage. So we shouldn't have this. Makes sense? 

Probably we can keep this entry in the `PROJECT` file and remove the unused functions from the byocluster_webhook file. What do you think?

Reply:
@anusha94 ok.

Reply:
You will have to delete just these
```
var _ webhook.Defaulter = &ByoCluster{}

// Default implements webhook.Defaulter so a webhook will be registered for the type
// nolint: stylecheck
func (r *ByoCluster) Default() {
}
```
Okay fine let's keep it. No harm I guess.

Reply:
Good, thanks.